File: foxy/parent/src/command/vanilla/economy/DailyExecutor.ts
import { ButtonStyles, MessageComponentTypes } from 'discordeno/types';
import { bot } from '../../../FoxyLauncher';
import { MessageFlags } from '../../../utils/discord/Message';
import { colors } from '../../../../../../common/utils/colors';
import { ExecutorParams } from '../../structures/CommandExecutor';

export default class DailyExecutor {

    async execute({ context, endCommand, t }: ExecutorParams) {
        context.reply({
            embeds: [{
                color: colors.FOXY_DEFAULT,
                title: context.makeReply(bot.emotes.FOXY_DAILY, t('commands:daily.embed.title')),
                description: context.makeReply(bot.emotes.FOXY_YAY, t('commands:daily.embed.description')),
                thumbnail: {
                    url: bot.foxy.constants.DAILY_EMOJI
                }
            }],
            components: [{
                type: MessageComponentTypes.ActionRow,
                components: [{
                    type: MessageComponentTypes.Button,
                    style: ButtonStyles.Link,
                    label: t('commands:daily.embed.button'),
                    emoji: {
                        id: BigInt(bot.emotes.FOXY_PETPET)
                    },
                    url: bot.foxy.constants.DAILY
                },
                {
                    type: MessageComponentTypes.Button,
                    style: ButtonStyles.Link,
                    label: t('commands:daily.embed.buyMore'),
                    emoji: {
                        id: BigInt(bot.emotes.FOXY_DAILY)
                    },
                    url: bot.foxy.constants.PREMIUM
                }]
            }],
            flags: MessageFlags.EPHEMERAL
        });

        return endCommand();
    }
}

File: foxy/src/main/kotlin/net/cakeyfox/foxy/interactions/vanilla/economy/DailyExecutor.kt
package net.cakeyfox.foxy.interactions.vanilla.economy

import net.cakeyfox.common.Colors
import net.cakeyfox.common.Constants
import net.cakeyfox.common.FoxyEmotes
import net.cakeyfox.foxy.interactions.commands.CommandContext
import net.cakeyfox.foxy.interactions.commands.UnleashedCommandExecutor
import net.cakeyfox.foxy.utils.linkButton
import net.cakeyfox.foxy.interactions.pretty

class DailyExecutor : UnleashedCommandExecutor() {
    override suspend fun execute(context: CommandContext) {
        context.reply(true) {
            embed {
                title = pretty(
                    FoxyEmotes.FoxyDaily,
                    context.locale["daily.embed.title"]
                )

                thumbnail = Constants.DAILY_EMOJI
                description = context.locale["daily.embed.description"]
                color = Colors.FOXY_DEFAULT

                actionRow(
                    linkButton(
                        FoxyEmotes.FoxyPetPet,
                        context.locale["daily.embed.redeemDaily"],
                        Constants.DAILY
                    ),

                    linkButton(
                        FoxyEmotes.FoxyDaily,
                        context.locale["daily.embed.buyMore"],
                        Constants.PREMIUM
                    )
                )
            }
        }
    }
}